integrations/llama_cpp/src/haystack_integrations/components/generators/llama_cpp/chat/chat_generator.py

```diff
@@ -122,14 +143,14 @@ def __init__(
         model_kwargs.setdefault("n_ctx", n_ctx)
         model_kwargs.setdefault("n_batch", n_batch)
 
-        _check_duplicate_tool_names(tools)
+        _check_duplicate_tool_names(list(tools or []))
```

Comment:
I may be wrong, but if tools is a `Toolset`, doesn't `list(tools)` create a nested structure?